playbook/actions/pod.go

```diff
@@ -0,0 +1,116 @@
+package actions
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/flanksource/commons/logger"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"github.com/flanksource/incident-commander/api"
+	v1 "github.com/flanksource/incident-commander/api/v1"
+)
+
+const (
+	defaultMaxLength        = 3000
+	defaultContainerTimeout = 5 * time.Minute
+
+	sidecarContainerName     = "container-waiter"
+	containerImage           = "ubuntu:jammy"
+	playbookActionSelector   = "playbooks.mission-control.flanksource.com/actions"
+	playbookActionLabelValue = "playbooks-pod-action"
+)
+
+type PodResult struct {
+	Logs string
+}
+
+type Pod struct {
+}
+
+func (c *Pod) Run(ctx api.Context, action v1.PodAction, env TemplateEnv) (*PodResult, error) {
+	timeout := time.Duration(action.Timeout) * time.Minute
+	if timeout == 0 {
+		timeout = defaultContainerTimeout
+	}
+
+	if action.MaxLength <= 0 {
+		action.MaxLength = defaultMaxLength
+	}
+
+	pod, err := newPod(ctx, action)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := cleanupExistingPods(ctx, fmt.Sprintf("%s=%s", playbookActionLabelValue, pod.Labels[playbookActionLabelValue])); err != nil {
```

Comment:
I am not sure we want to cleanup existing pods, during a playbook execution - Considering we may run the same step many times for different inputs. 

The pod name should probably be generated from the run_step_id, and then get labelled with the playbook name, id, stepname and step id and then have seperate job to cleanup orphans - maybe daily